### ISSUE TITLE
extraction using "-x" and aux-file stuck

### ISSUE THREAD

**Issue by tutebatti** [ge-ne/bibtool#81]
Hello,
using -x (or extract.file in a rsc-file), nothing happens. Verbose output via -v option shows as last line:

BibTool: Reading <stdin>

Am I missing something?

Thanks a lot!

**Comment by ge-ne** [comment]
It appears that you have not specified any input file. In this case BibTool reads from the standard input stream...

**Comment by tutebatti** [comment]
I provided an aux-file as written in the documentation:
`bibtool -x bib.aux -o extract.bib`
Alternatively, in a rsc-file:
`extract.file = {bib.aux}`
I understood that no additional bib-file as input is needed.

**Comment by ge-ne** [comment]
Right. The input file(s) need to be given in the aux file. It must contain one or more \bibdata lines. In this case those files are read. Otherwise BibTool falls back to reading from stdin.

The \bibdata is written to the aux file by the LaTeX \bibliography macro

I am pretty sure that your aux file does not contain such lines.

**Comment by jpgoldberg** [comment]
I suspect that this is a BiBTeX versus biber issue. I experienced the same problem, and indeed my `.aux` file does not contain any instances of `\bibdata`.

The good news is that the `.bbl` file that is generated by biber may be sufficient for most of the purposes one might want to  generate a `.bib` file from an aux file. It will contain all the bibliographic data needed for all of the cited works.

**Comment by jpgoldberg** [comment]
@tutebatti, it appears that the best tool for you (and I and other biblatex users) is bbl2bib, which should be part of a TeXlive setup.

**Comment by tutebatti** [comment]
@jpgoldberg Thank you. Could you explain how exactly this helps in using the -x option of bibtool?

**Comment by CodePurble** [comment]
`biber` itself can be used to do what you want. The command would be:

```sh
biber texfile --output-format bibtex
```

> Assuming the LaTeX file is texfile.tex
> This is tested for `biber` version 2.16
